### PR TITLE
Print errors to stderr instead of stdout

### LINKED ISSUE #75: Print errors to stderr

### PR DESCRIPTION
**Pull request by vsham20**
Fixes #75.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -76,12 +76,12 @@ fn main() {
         build.include(include);
     }
 
-    println!("cargo:rustc-link-search=native={}", get_llvm_libdir());
+    eprintln!("cargo:rustc-link-search=native={}", get_llvm_libdir());
```
Comment (marco-c) [inline build.rs]: These are not errors, so we should keep using println! here

File: src/gcov.rs [hunk 2]
```diff
@@ -21,16 +21,16 @@ fn gcov_read_unsigned() -> u32 {
 
 fn prova() {
   if gcov_open("/home/marco/Documenti/workspace/grcov/tests/llvm/main.gcda".to_string()) == 1 {
-    println!("2");
```
Comment (marco-c) [inline src/gcov.rs]: Same here and in the following cases in this file

File: src/main.rs [hunk 3]
```diff
@@ -25,22 +25,22 @@ macro_rules! println_stderr(
 );
 
 fn print_usage(program: &str) {
-    println!("Usage: {} DIRECTORY_OR_ZIP_FILE[...] [-t OUTPUT_TYPE] [-s SOURCE_ROOT] [-p PREFIX_PATH] [--token COVERALLS_REPO_TOKEN] [--commit-sha COVERALLS_COMMIT_SHA] [--keep-global-includes] [--ignore-not-existing] [--ignore-dir DIRECTORY] [--llvm] [--path-mapping PATH_MAPPING_FILE] [--branch]", program);
```
Comment (marco-c) [inline src/main.rs]: This is the usage text, not an error, so we should keep printing it with println!

File: src/output.rs [hunk 4]
```diff
@@ -30,7 +30,7 @@ pub fn output_activedata_etl(results: CovResultIter) {
         start_indexes.sort();
 
         for (name, function) in &result.functions {
-            // println!("{} {} {}", name, function.executed, function.start);
+            // eprintln!("{} {} {}", name, function.executed, function.start);
```
Comment (marco-c) [inline src/output.rs]: Not an error

File: src/output.rs [hunk 5]
```diff
@@ -55,7 +55,7 @@ pub fn output_activedata_etl(results: CovResultIter) {
                 orphan_uncovered.remove(line);
             }
 
-            println!("{}", json!({
+            eprintln!("{}", json!({
```
Comment (marco-c) [inline src/output.rs]: This is the output of the program, it's not an error

File: src/output.rs [hunk 6]
```diff
@@ -75,7 +75,7 @@ pub fn output_activedata_etl(results: CovResultIter) {
         let orphan_uncovered: Vec<u32> = orphan_uncovered.into_iter().collect();
 
         // The orphan lines will represent the file as a whole.
-        println!("{}", json!({
+        eprintln!("{}", json!({
```
Comment (marco-c) [inline src/output.rs]: Same here

File: src/output.rs [hunk 7]
```diff
@@ -104,7 +104,7 @@ pub fn output_lcov(results: CovResultIter) {
     writer.write_all(b"TN:\n").unwrap();
 
     for (_, rel_path, result) in results {
-        // println!("{} {:?}", rel_path, result.lines);
+        // eprintln!("{} {:?}", rel_path, result.lines);
```
Comment (marco-c) [inline src/output.rs]: Not an error

File: src/parser.rs [hunk 8]
```diff
@@ -230,7 +230,7 @@ pub fn parse_gcov(gcov_path: &Path) -> Result<Vec<(String,CovResult)>, ParserErr
         match key {
             "file" => {
                 if cur_file.is_some() && !cur_lines.is_empty() {
-                    // println!("{} {} {:?}", gcov_path.display(), cur_file, cur_lines);
+                    // eprintln!("{} {} {:?}", gcov_path.display(), cur_file, cur_lines);
```
Comment (marco-c) [inline src/parser.rs]: Not an error

File: test/rust/generics_with_two_parameters_old.gcov [hunk 9]
```diff
@@ -7,8 +7,8 @@
         -:    2:
 function _ZN27rust_code_coverage_sample_244compare_types<[i32; 3],alloc::vec::Vec<i32>>E called 1 returned 100% blocks executed 100%
         -:    3:fn compare_types<T: Debug, U: Debug>(t: &T, u: &U) {
-        3:    4:    println!("t: `{:?}", t);
```
Comment (marco-c) [inline test/rust/generics_with_two_parameters_old.gcov]: This is a test support file, it isn't source code

File: tests/test.rs [hunk 10]
```diff
@@ -140,7 +140,7 @@ fn check_equal_coveralls(expected_output: &str, output: &str, skip_branches: boo
     let expected: Value = serde_json::from_str(expected_output).unwrap();
     let actual: Value = serde_json::from_str(output).unwrap();
 
-    println!("{}", serde_json::to_string_pretty(&actual).unwrap());
+    eprintln!("{}", serde_json::to_string_pretty(&actual).unwrap());
```
Comment (marco-c) [inline tests/test.rs]: Not an error

File: tests/test.rs [hunk 11]
```diff
@@ -221,15 +221,15 @@ fn test_integration() {
...
+            eprintln!("\n\n{}", path.display());
 
             let skip_branches = path == Path::new("tests/template") || path == Path::new("tests/include") ||
                                 path == Path::new("tests/include2") || path == Path::new("tests/class");
 
             make_clean(path);
 
             if !cfg!(windows) {
-                println!("GCC");
+                eprintln!("GCC");
```
Comment (marco-c) [inline tests/test.rs]: Not an error

File: tests/test.rs [hunk 12]
```diff
@@ -238,7 +238,7 @@ fn test_integration() {
             }
 
             if !cfg!(target_os="macos") {
-                println!("\nLLVM");
+                eprintln!("\nLLVM");
```
Comment (marco-c) [inline tests/test.rs]: Not an error

**Comment by vsham20** [comment]
@marco-c yeah sorry, Editor messed up my code, didn't use it correct way :)
But I have updated the  PR, please have a look.

**Comment by marco-c** [comment]
Could you also remove the `println_stderr` macro definition from src/main.rs and replace its usage with `eprintln`?

**Comment by vsham20** [comment]
@marco-c Thanks :)

**Pull request merged**